app/controllers/spree/admin/videos_controller.rb

```diff
@@ -44,24 +60,52 @@ def update
       # DELETE /videos/1
       def destroy
         @video.destroy
-        redirect_to videos_url, notice: 'Video was successfully destroyed.'
+        if @video.destroy
+          flash[:success] = 'Video was successfully destroyed.'
+        else
+          flash[:error] = 'Video was not successfully destroyed.'
+        end
+        respond_with(@video) do |format|
+          format.html { redirect_to collection_url }
+          format.js { render_js_for_destroy }
+        end
       end
 
       private
         # Use callbacks to share common setup or constraints between actions.
+        def collection
+          params[:q] = {} if params[:q].blank?
+          videos = super.order(priority: :asc)
+          @search = videos.ransack(params[:q])
+
+          @collection = @search.result.
+              page(params[:page]).
+              per(params[:per_page])
+        end
+
         def set_video
           @video = Video.find(params[:id])
         end
 
         # Only allow a list of trusted parameters through.
         def video_params
-          params.fetch(:video, {}).permit(:name,:slug,:description,:seo_title,:seo_description, :primary_product, product_ids: [])
+          params.require(:video).permit(:vendor_id, :name,:slug,:description,:seo_title,:seo_description,:available_on,:discontinue_on,:primary_product_id,:upload_video, product_ids: [], taxon_ids:[] )
         end
 
-        def load_data
-          @available_products = Product.order(:name)
+        def scope
+          Spree::Video.accessible_by(current_ability)
         end
 
+        def load_data
+          if current_spree_user.has_spree_role?("admin")
+            @products = Product.order(:name)          
+          else
+            @products = Product.order(:name).where(vendor_id: current_spree_user.vendors.active.ids) 
```

Comment:
The logic they have used for products listing the same logic i have used for videos listing, I can try updating this method but anyway video extension is  dependent on multi vendors since it has vendor drop down. so to make it independent  i will have to add vendor drop down and other logics from multi vendor extension.

Reply:
ok, we will figure this later then. in case there is any requirement of videos without vendors